packages/eas-cli/src/utils/code-signing.ts

```diff
@@ -4,17 +4,59 @@ import {
   signStringRSASHA256AndVerify,
   validateSelfSignedCertificate,
 } from '@expo/code-signing-certificates';
+import { ExpoConfig } from '@expo/config';
 import Dicer from 'dicer';
 import isDeepEqual from 'fast-deep-equal';
 import { promises as fs } from 'fs';
 import { Response } from 'got';
 import { pki as PKI } from 'node-forge';
 import nullthrows from 'nullthrows';
+import path from 'path';
 import { Stream } from 'stream';
 import { parseItem } from 'structured-headers';
 
 import { PartialManifest, PartialManifestAsset } from '../graphql/generated';
 
+type CodeSigningInfo = {
+  privateKey: PKI.rsa.PrivateKey;
+  certificate: PKI.Certificate;
+  codeSigningMetadata: { alg: string; keyid: string };
+};
+
+export async function getCodeSigningInfoAsync(
+  config: ExpoConfig,
+  privateKeyPath: string | undefined
+): Promise<CodeSigningInfo | undefined> {
+  const codeSigningCertificatePath = (config.updates as any)?.codeSigningCertificate;
+  const codeSigningMetadata = (config.updates as any)?.codeSigningMetadata;
+
+  if (codeSigningCertificatePath && !privateKeyPath) {
+    privateKeyPath = path.join(path.dirname(codeSigningCertificatePath));
+  }
+
+  if (!codeSigningMetadata || !codeSigningMetadata.alg || !codeSigningMetadata.keyid) {
+    throw new Error('Must specify codeSigningMetadata in config.updates for EAS code signing');
```

Comment:
This sounds like there is a field called "config":
```suggestion
    throw new Error('Must specify codeSigningMetadata under the "updates" field of your app config file to use EAS code signing');
```